l10n_nl_intrastat/models/l10n_nl_intrastat.py

```diff
@@ -0,0 +1,220 @@
+    _order = "date_to desc"
+    _rec_name = 'date_range_id'
+    _inherit = ['intrastat.common']
+
+    last_updated = fields.Datetime(readonly=True, required=False)
+    date_range_id = fields.Many2one(
+        comodel_name='date.range',
+        required=True,
+        string='Date range'
+    )
+    date_from = fields.Date(related='date_range_id.date_start', readonly=True)
+    date_to = fields.Date(related='date_range_id.date_end', readonly=True,
+            store=True)
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        default=lambda self: self.env.user.company_id,
+        string='Company'
+    )
+    total_amount = fields.Float(
+        string='Total amount',
+        readonly=True,
+        help="Total amount in company currency of the declaration.",
+        digits=dp.get_precision('Account'),
+        # compute='_compute_lines_and_total'
+    )
+    state = fields.Selection(
+        string='State',
+        selection=[
+            ('draft', 'Draft'),
+            ('done', 'Done'),
+        ],
+        default='draft',
+        readonly=True,
+        help=(
+            "State of the declaration. When the state is set to 'Done', "
+            "the parameters become read-only."
+        ),
+    )
+    line_ids = fields.One2many(
+        string='ICP line',
+        comodel_name='l10n_nl.report.intrastat.line',
+        inverse_name='report_id',
+        readonly=True,
+        # compute='_compute_lines_and_total'
+    )
+
+    @api.multi
+    def set_draft(self):
+        """ Reset report state to draft. """
+        return self.write({'state': 'draft'})
+
+    @api.multi
+    def set_done(self):
+        """ Validate and close report. """
+        return self.write({'state': 'done'})
+
+    @api.multi
+    def generate_lines(self):
+        """
+        Collect the data lines for the given report.
+        Unlink any existing lines first.
+        """
+        self.ensure_one()
+        # Other models:
+        invoice_model = self.env['account.invoice']
+        invoice_line_model = self.env['account.invoice.line']
+
+        # Check whether all configuration done to generate report
+        self._check_generate_lines()
+
+        # We need at least these variables to calculate
+        if not self.date_from or not self.date_to or not self.company_id:
+            return
+
+        # Define search for invoices for period and company:
+        company_obj = self.company_id  # simplify access
+        invoice_domain = [
+            ('type', 'in', ('out_invoice', 'out_refund')),
+            ('date_invoice', '>=', self.date_from),
+            ('date_invoice', '<=', self.date_to),
+            ('state', 'in', ('open', 'paid')),
+            ('company_id', '=', self.company_id.id),
+        ]
+        # Signal invoices without a country
+        # (Should not happen, as country_id on parter is set to required)
+        invalid_invoices = invoice_model.search_read(
```

Comment:
It would make the module more friendly to implement in client installs, where there are typically only a few foreign partners and people don't want to always select country. Do you know why this wasn't done in the first place?